compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java

```diff
@@ -616,6 +618,7 @@ private void rewriteObservableFunctionInvocations(BIRFunction func, BIRPackage p
                         BIRBasicBlock observeEndBB = insertBasicBlock(func, eeTargetIndex + 1);
                         BIRBasicBlock newTargetBB = insertBasicBlock(func, eeTargetIndex + 2);
                         swapBasicBlockContent(errorEntry.targetBB, newTargetBB);
+                        resetLocalVarEndBBs(func, errorEntry.targetBB, newTargetBB);
```

Comment:
Does it make more sense to move this into the `swapBasicBlockContent` method itself ?
(Also there is another `swapBasicBlockTerminator` method as well where we only swap the terminator.)

Reply:
I think when we have an observability terminator (eg:- HTTP Client call), the the immediate local vars might not be visible when not swapped. But it's not the exact bug we are trying to fix I think.

Reply:
Yes, adding this to when we do the terminator swap only is not be relevant to the fix for this particular issue. IMO reassigning endBBs of local variables after the `Return` or `Panic` terminator swap may not be useful. As you mentioned, when the terminator instruction is `Call` and it is observable, the endBBs may need to be reassigned after the terminator swap. I'll make changes accordingly.

Reply:
Addressed with https://github.com/ballerina-platform/ballerina-lang/pull/32667/commits/3d12d91fd7257fcb7f4417157a159d0be72b84cd

Reply:
Yes for return and panic it's probably not required 👍🏻